client/src/pages/Login.jsx

```diff
@@ -1,50 +1,74 @@
 import React, { useState } from "react";
 import axios from "axios";
-import { Box, Typography, TextField, Button } from "@mui/material";
-import { Link, useNavigate } from "react-router-dom";
+import { Box, Typography, TextField, Button, Paper } from "@mui/material";
+import { useNavigate } from "react-router-dom";
+
 export const Login = () => {
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
   const [userName, setUserName] = useState("");
   const navigate = useNavigate();
+
   async function loginUser(e) {
     e.preventDefault();
-    const response = await axios.post(
-      "http://localhost:4000/api/login",
-      {
-        email,
-        password,
-      },
-      {
-        headers: {
-          "Content-Type": "application/json",
-        },
-      }
-    );
+    try {
+      const response = await axios.post(
+        "http://localhost:4000/api/login",
+        { email, password },
+        { headers: { "Content-Type": "application/json" } }
+      );
+      const data = response.data;
+      const token = data.token;
 
-    const data = response.data;
-    const token = response.data.token;
-    if (token) {
-      console.log("User Logged Successfully");
-      setUserName(data.name);
-      navigate("/dashboard", { state: { userName: data.name } });
-    } else {
-      console.log("Unauthorized User");
-      alert("Login Failed: Invalid Credentials");
-      setEmail("");
-      setPassword("");
+      if (token) {
+        console.log("User Logged Successfully");
+        setUserName(data.name);
+        navigate("/dashboard", { state: { userName: data.name, token: data.token } });
+      } else {
+        console.log("Unauthorized User");
+        alert("Login Failed: Invalid Credentials");
+        setEmail("");
+        setPassword("");
+      }
+      console.log(data);
+    } catch (error) {
+      console.error("Login error:", error);
+      alert("Login failed: Please check your credentials and try again.");
     }
-    console.log(data);
   }
+
   function handleRegister() {
     navigate("/register");
   }
 
   return (
-    <>
-      <>
-        <Typography variant="h1" component="h2" gutterBottom>
-          Login
+    <Box
+      sx={{
+        height: "100vh",
+        display: "flex",
+        justifyContent: "center",
+        alignItems: "center",
+        backgroundImage: `url('https://source.unsplash.com/1600x900/?login')`,
```

Comment:
Check if this thing is working or not, same in the register and homepage as well, where this is being used in the background image, check if this is loading or not